06-building-a-program-start-to-finish/solving-the-problem/output/app/class/invertedIndexClass.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * Inverted Index class
+ * @class
+ */
+class InvertedIndex {
+
+  /**
+   * class constructor
+   * @constructor
+  */
+  constructor() {
+    this.indices = {};
+    this.indexedFiles = {};
+  }
+
+  /**
+   * function to read uploaded files
+   * @function
+   * @param {object} file
+   * @returns {Promise} promise
+  */
+  static readFile(file) {
+    const fileReader = new FileReader();
+    fileReader.readAsText(file);
+
+    return new Promise((resolve, reject) => {
+      fileReader.onload = () => {
+        try {
+          return resolve(JSON.parse(fileReader.result));
+        } catch (e) {
+          return reject('improperly formatted JSON file');
+        }
+      };
+    });
+  }
+
+  /**
+   * function to genereate indexes from read files
+   * @function
+   * @param {Array} fileName
+   * @param {string} file
+   * @returns {boolean} returns a boolean
+  */
+  createIndex(fileName, file) {
+    if (!file) { return false; }
+
+    const words = [];
+    const indexedWords = {};
+    file.forEach((book) => {
+      words.push(InvertedIndex.tokenize(book));
+    });
+
+    words.forEach((book, index) => {
+      for (let i = 0; i < book.length; i += 1) {
+        if (!Object.prototype.hasOwnProperty.call(indexedWords, book[i])) {
+          indexedWords[book[i]] = [];
+        }
+        indexedWords[book[i]].push(index);
+      }
+    });
+
+    if (!Object.prototype.hasOwnProperty.call(this.indices, fileName)) {
+      this.indices[fileName] = indexedWords;
+    }
+    this.indexedFiles[fileName] = file.length;
+    return true;
+  }
+
+  /**
+   * function to fetch generated indices
+   * @function
+   * @param {string} fileName
+   * @returns {Object} indices
+  */
+  getIndices(fileName) {
+    return typeof fileName === 'undefined'
+    || typeof fileName !== 'string' ? this.indices : this.indices[fileName];
+  }
+
+  /**
+   * function to search through created indices
+   * @function
+  * @param {String} fileName uploaded valid JSON file
+  * @param {String} query word(s) or terms to search for
+  * @returns {array} returns array searchResults contained search indices
+  */
+  searchIndices(fileName, query) {
```

Comment:
Are you really passing in a query or key-words or search-terms you want to search for?